test/common/upstream/eds_test.cc

```diff
@@ -1456,6 +1456,65 @@ TEST_F(EdsTest, EndpointLocality) {
   EXPECT_EQ(nullptr, cluster_->prioritySet().hostSetsPerPriority()[0]->localityWeights());
 }
 
+// Validate that onConfigUpdate() updates the endpoint locality of an existing endpoint.
+TEST_F(EdsTest, EndpointLocalityUpdated) {
```

Comment:
Can you add a test that verifies that the runtime flag can be used to revert to the old behavior? Thanks

Reply:
done. PTAL!